libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt

```diff
@@ -99,11 +105,21 @@ class RouteOptionsUpdater {
                     .snappingIncludeClosuresList(
                         let snappingClosures@{
                             val snappingClosures = routeOptions.snappingIncludeClosuresList()
-                            if (snappingClosures.isNullOrEmpty()) {
-                                return@snappingClosures emptyList<Boolean>()
-                            }
-                            mutableListOf<Boolean>().also {
-                                it.addAll(snappingClosures.subList(index, coordinatesList.size))
+                            mutableListOf<Boolean?>().apply {
```

Comment:
this implicitly changes the behavior of the method, even if it's not used for re-route aims. What if move it to `MapboxRerouteController` similar to extension `RouteOptions?.applyRerouteOptions`

Reply:
I think that's okay. The change we're making here is a clear improvement, we haven't identified an instance where `snapping_include_closures` shouldn't be used for an updated, mid-route request.